Sources/Sentry/SentryTracer.m

```diff
@@ -40,7 +41,11 @@
  */
 static const NSTimeInterval SENTRY_APP_START_MEASUREMENT_DIFFERENCE = 5.0;
 static const NSTimeInterval SENTRY_AUTO_TRANSACTION_MAX_DURATION = 500.0;
+#if SENTRY_DEVELOPMENT
+static const NSTimeInterval SENTRY_AUTO_TRANSACTION_DEADLINE = 5.0;
+#else
 static const NSTimeInterval SENTRY_AUTO_TRANSACTION_DEADLINE = 30.0;
+#endif
```

Comment:
`m`: Why do we need this?

Reply:
Make sample tests while developing so much easier. Dont need to wait 30 seconds to see deadline timeout while debugging.

Reply:
I would just change the value to 5.0 when you are debugging. This could cause quite some confusion if you are unaware of that.

Reply:
What kind of confusion?
We have debug logs to inform that the deadline expired. If one of us see something finishing early it will be clear why.

Reply:
It could just be `DEBUG` for now though, since it's only defined in the Debug build config along with `DEBUG`. If we only treated warnings as errors for nondebug builds, I'd say we could add `#warning "Using shorter auto transaction deadlines"` here as well.

I agree this would be confusing during unit testing. If you're only using it for testing with a sample app, we could instead add a way to configure it in the sample app's UI.

I wondered if it's something we could open up for config in SentryOptions but that probably has backend/product implications and I assume that'll be a nonstarter.

Reply:
If we use DEBUG it will also run like this in customers projects

Reply:
> If we use DEBUG it will also run like this in customers projects

That's another reason for me to remove this. It confuses customers.